n3fit/src/n3fit/layers/msr_normalization.py

```diff
@@ -27,20 +56,23 @@ def __init__(self, output_dim=14, mode="ALL", **kwargs):
         else:
             raise ValueError(f"Mode {mode} not accepted for sum rules")
 
-        idx = []
+        self.indices = []
+        self.divisor_indices = []
         if self._msr_enabled:
-            idx += GLUON_IDX
+            self.indices += [IDX[c] for c in MSR_COMPONENTS]
+            self.divisor_indices += [IDX[MSR_DENOMINATORS[c]] for c in MSR_COMPONENTS]
         if self._vsr_enabled:
-            idx += V_IDX + V3_IDX + V8_IDX + V15_IDX
-
-        self._out_scatter = op.as_layer(
-            op.scatter_to_one, op_kwargs={"indices": idx, "output_dim": output_dim}
-        )
+            self.divisor_indices += [IDX[VSR_DENOMINATORS[c]] for c in VSR_COMPONENTS]
+            self.indices += [IDX[c] for c in VSR_COMPONENTS]
+            self.vsr_factors = op.numpy_to_tensor([VSR_CONSTANTS[c] for c in VSR_COMPONENTS])
+        # Need this extra dimension for the scatter_to_one operation
+        self.indices = [[i] for i in self.indices]
```

Comment:
It would be better to remove `self.` from the previous `self.indices` since you are doing this at the end (`indices` being shape `(n,)` and `self.indices` shape `(n,1)`)